<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TrunkDetectionForm</class>
 <widget class="QWidget" name="TrunkDetectionForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>285</width>
    <height>719</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabs_detection">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab_bottom_up">
      <attribute name="title">
       <string>Bottom-up</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="group_height_filter">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>100</height>
          </size>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="title">
          <string>Height filter</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="MinMaxWidget" name="minmax_height" native="true">
            <property name="singleStep" stdset="0">
             <double>0.500000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="group_euclidean_clustering">
         <property name="title">
          <string>Euclidean clustering</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_horizontal_threshold">
            <property name="text">
             <string>Horizontal threshold</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="spinbox_threshold_horizontal">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.001000000000000</double>
            </property>
            <property name="maximum">
             <double>10.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.600000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_vertical_threshold">
            <property name="text">
             <string>Vertical threshold</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="spinbox_threshold_vertical">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.001000000000000</double>
            </property>
            <property name="maximum">
             <double>10.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.400000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_cluster_tolerance">
            <property name="text">
             <string>Cluster tolerance</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="spinbox_cluster_tolerance">
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="maximum">
             <double>50.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>8.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_min_cluster_size">
            <property name="text">
             <string>Minimum cluster size</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="spinbox_min_cluster_size">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_top_down">
      <attribute name="title">
       <string>Top-down</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_radius">
         <property name="text">
          <string>Radius</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="spinbox_radius">
         <property name="decimals">
          <number>1</number>
         </property>
         <property name="minimum">
          <double>0.100000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>1.500000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="button_detect">
     <property name="text">
      <string>Detect</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="group_clusters">
     <property name="title">
      <string>Clusters</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QListView" name="list_clusters">
        <property name="flow">
         <enum>QListView::LeftToRight</enum>
        </property>
        <property name="spacing">
         <number>2</number>
        </property>
        <property name="viewMode">
         <enum>QListView::IconMode</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="button_enrich">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Enrich</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_use_as_seeds">
          <property name="text">
           <string>Use as seeds</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="group_line_fitting">
     <property name="title">
      <string>Line fitting</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_angular_threshold">
        <property name="text">
         <string>Angular threshold</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="spinbox_angular_threshold">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>90</number>
        </property>
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_distance_threshold">
        <property name="text">
         <string>Distance threshold</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="spinbox_distance_threshold">
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>5.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.200000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QPushButton" name="button_fit">
        <property name="text">
         <string>Fit</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MinMaxWidget</class>
   <extends>QWidget</extends>
   <header>min_max_widget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_detect</sender>
   <signal>clicked()</signal>
   <receiver>TrunkDetectionForm</receiver>
   <slot>onButtonDetectClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>135</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_use_as_seeds</sender>
   <signal>clicked()</signal>
   <receiver>TrunkDetectionForm</receiver>
   <slot>onButtonUseAsSeedsClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>535</y>
    </hint>
    <hint type="destinationlabel">
     <x>135</x>
     <y>286</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinbox_angular_threshold</sender>
   <signal>valueChanged(int)</signal>
   <receiver>TrunkDetectionForm</receiver>
   <slot>onButtonFitClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>422</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>307</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinbox_distance_threshold</sender>
   <signal>valueChanged(double)</signal>
   <receiver>TrunkDetectionForm</receiver>
   <slot>onButtonFitClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>213</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>307</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_enrich</sender>
   <signal>clicked()</signal>
   <receiver>TrunkDetectionForm</receiver>
   <slot>onButtonEnrichClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>645</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_fit</sender>
   <signal>clicked()</signal>
   <receiver>TrunkDetectionForm</receiver>
   <slot>onButtonFitClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>146</x>
     <y>491</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onPassthroughEditingFinished()</slot>
  <slot>onButtonDetectClicked()</slot>
  <slot>onDensityAnalysisEditingFinished()</slot>
  <slot>onButtonUseAsSeedsClicked()</slot>
  <slot>onButtonFitClicked()</slot>
  <slot>onButtonEnrichClicked()</slot>
 </slots>
</ui>
